drivers/include/wiegand.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2019 Hackerspace San Salvador
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_wiegand Wiegand
+ * @ingroup     drivers_sensors
+ * @ingroup     drivers_saul
+ *
+ * @brief       GPIO based Wiegand driver
+ *
+ * This driver provides @ref drivers_saul capabilities.
+ * @{
+ *
+ * @file
+ * @brief       Driver for the Wiegand Protocol
+ *
+ * @author      Mario Gomez <mario.gomez@teubi.co>
+ */
+
+#ifndef WIEGAND_H
+#define WIEGAND_H
+
+#include <stdint.h>
+#include "periph/gpio.h"
+
+#ifdef __cplusplus
+extern "C" {
+  #endif
+
+  /**
+   * @brief   Parameters needed for device initialization
+   */
+  typedef struct {
+    gpio_t       d0;      /**< Wiegand D0 data line */
+    gpio_t       d1;      /**< Wiegand D1 data line */
+    gpio_flank_t flank;   /**< flank detection */
+  } wiegand_params_t;
+
+  /**
+   * @brief   Device descriptor for a wiegand device
+   */
+  typedef struct {
+    uint32_t code;                         /**< Wiegand code */
+    int16_t type;                          /**< Wiegand type */
+    volatile uint32_t wg_card_temp_high;   /**< Card MSB */
+    volatile uint32_t wg_card_temp;        /**< Card LSV  */
+    volatile uint32_t wg_last_wiegand;     /**< Last card lecture */
+    volatile int16_t wg_bit_count;         /**< Number of bits read */
```

Comment:
Why are these `volatile`?

Reply:
Those variables represent the wiegand buffer (*_card_temp), the last timestamp and number of bits for a wiegand frame. Those variables are modified within interrupt context.

However... After testing a little bit more I just found that **xtimer_now_usec64** is not returning a monotonic value if called within the pin change handler.

It is ok to let this "internal state" variables within the device structure? What's the best strategy to store internal state for drivers in RIOT? (I'm just trying to avoid globals)

Reply:
Sure the driver struct is where you keep global state for the driver.
This way you can have multiple instances of the driver.

Reply:
Done. Included in the latest commit.